whetstone/navigation/src/main/java/com/freeletics/mad/whetstone/internal/NavEntryViewModelProvider.kt

```diff
@@ -7,34 +7,35 @@ import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.createSavedStateHandle
 import androidx.lifecycle.viewmodel.initializer
 import androidx.lifecycle.viewmodel.viewModelFactory
-import androidx.navigation.NavBackStackEntry
 import com.freeletics.mad.navigator.BaseRoute
+import com.freeletics.mad.navigator.internal.NavigationExecutor
 import kotlin.reflect.KClass
 
 /**
- * Creates a [ViewModel] for the given [entry]. The `ViewModel.Factory` will use [scope] to lookup
- * a parent component instance. That component will then be passed to the given [factory] together
- * with a [SavedStateHandle] and the passed in [route].
+ * Creates a [ViewModel] for the given [destination]. The `ViewModel.Factory` will use [parentScope]
+ * to lookup a parent component instance. That component will then be passed to the given [factory]
+ * together with a [SavedStateHandle] and the passed in [destination].
  *
  * To be used in generated code.
  */
 @InternalWhetstoneApi
-public inline fun <reified T : ViewModel, D : Any, R : BaseRoute> viewModel(
-    entry: NavBackStackEntry,
+public inline fun <reified T : ViewModel, D : Any, R : BaseRoute> navEntryViewModel(
+    destination: KClass<R>,
```

Comment:
We use this for looking up both the route (doesn't need to happen in generated code) and the viewModelStore (removes the need for `NavBackStackEntry`)